src/test/kotlin/com/terraformation/backend/accelerator/SubmissionServiceTest.kt

```diff
@@ -0,0 +1,221 @@
+        )
+      }
+
+      val submissionDocument = submissionDocumentStore.fetchOneById(submissionDocumentId)
+      assertEquals(
+          submissionDocument,
+          ExistingSubmissionDocumentModel(
+              id = submissionDocumentId,
+              documentStore = DocumentStore.Google,
+              description = mockDocumentDescription,
+              location = mockStoredFileLocation,
+              name = expectedFileName,
+              submissionId = SubmissionId(1)))
+    }
+
+    @Test
+    fun `handles upload file name collisions correctly`() {
+      val projectId: ProjectId = insertProject()
+      insertProjectDocumentSettings(projectId)
+      val deliverableId: DeliverableId = insertDeliverable()
+
+      val deliverable = deliverableStore.fetchOneById(deliverableId)
+      val projectDocumentSettings = projectDocumentSettingsStore.fetchOneById(projectId)
+
+      val mockDocumentDescription = "The budget"
+      val mockStoredFileLocation = "test-location"
+
+      val expectedFileName1 = "Deliverable 1_1970-01-01_PHL_CCCO2_The budget.pdf"
+      val expectedFileName2 = "Deliverable 1_1970-01-01_PHL_CCCO2_The budget_2.pdf"
+
+      val mockStoredFile = StoredFile(expectedFileName1, "test-location")
+      every { googleDriveReceiver.upload(any(), any(), any()) } returns mockStoredFile
+
+      val submissionDocumentId1 =
+          receiveDocument(
+              deliverable = deliverable,
+              description = mockDocumentDescription,
+              projectDocumentSettings = projectDocumentSettings,
+              projectId = projectId,
+              originalName = "test-budget&?.pdf",
+          )
+      val submissionDocumentId2 =
+          receiveDocument(
+              deliverable = deliverable,
+              description = mockDocumentDescription,
+              projectDocumentSettings = projectDocumentSettings,
+              projectId = projectId,
+              originalName = "test-budget.pdf",
+          )
+
+      val submissionDocument1 = submissionDocumentStore.fetchOneById(submissionDocumentId1)
+      val submissionDocument2 = submissionDocumentStore.fetchOneById(submissionDocumentId2)
+
+      verify {
+        googleDriveReceiver.upload(
+            inputStream = inputStream,
+            fileName = expectedFileName1,
+            contentType = contentType,
+        )
+      }
+      verify {
+        googleDriveReceiver.rename(
+            storedFile = mockStoredFile,
+            newName = expectedFileName2,
+        )
+      }
+
+      assertEquals(
+          listOf(submissionDocument1, submissionDocument2),
+          listOf(
+              ExistingSubmissionDocumentModel(
+                  id = submissionDocumentId1,
+                  documentStore = DocumentStore.Google,
+                  description = mockDocumentDescription,
+                  location = mockStoredFileLocation,
+                  name = expectedFileName1,
+                  submissionId = SubmissionId(2)),
```

Comment:
This part is a bit iffy. I tried making the `setUp` function only run once with `BeforeAll` and `@TestInstance(TestInstance.Lifecycle.PER_CLASS)`, but ran into user auth issues. I think ideally the "create project, insert project document settings" only happens once for this entire test class. But I was not able to make it work, so the second test has _the next_ submission ID.

Perhaps there's something I am missing? Or maybe I need to delete the entries after each test? Or maybe I am overthinking it?

Reply:
Here's how to reset a table's ID sequence for each test method in a `DatabaseTest` subclass:
```kotlin
override val tablesToResetSequences = listOf(SUBMISSIONS)
```